product-scenarios/2-collaborative-api-development/2.1-manage-publisher-portal-operations-for-users/2.1.1-manage-api-publishing-according-to-the-publisher-roles-and-permission/src/test/java/org/wso2/am/scenario/tests/rest/api/publisherRoleAndPermission/PublishAPIByValidRolePermissionCategoryTestCase.java

```diff
@@ -0,0 +1,303 @@
+                new URL(backendEndPoint));
+        createAPI(apiRequest);
+        getAPI(apiName, developer, apiVersion);
+        apiNames.put(apiName, developer);
+        createUser(testUser, password, new String[]{role}, ADMIN_LOGIN_USERNAME, ADMIN_PASSWORD);
+        userList.add(testUser);
+        checkPublishAPI(state, role);
+        updateUser(developer, new String[]{"internal/subscriber"}, null, ADMIN_LOGIN_USERNAME, ADMIN_PASSWORD);
+        loginToStore(developer, password);
+        isAPIVisibleInStore(apiName, apiStoreClient);
+    }
+
+    @Test(description = "2.1.1.2", dataProvider = "ValidPermissionDataProvider",
+            dataProviderClass = ScenarioDataProvider.class)
+    public void testPublishAPIByValidPermissionUser(String[] permissionList) throws Exception {
+
+        apiName = "API_" + count;
+        apiContext = "/verify_" + count;
+        developer = "User_" + count;
+        userRole = "role" + count;
+        testUser = "User" + count;
+        password = "password123$";
+        count++;
+
+        createUserWithPublisherAndCreatorRole(developer, password, ADMIN_LOGIN_USERNAME, ADMIN_PASSWORD);
+        apiPublisher.login(developer, password);
+
+        apiRequest = new APIRequest(apiName, apiContext, apiVisibility, apiVersion, apiResource, tierCollection,
+                new URL(backendEndPoint));
+        createAPI(apiRequest);
+        getAPI(apiName, developer, apiVersion);
+        apiNames.put(apiName, developer);
+
+        createRole(ADMIN_LOGIN_USERNAME, ADMIN_PASSWORD, userRole, permissionList);
+        roleList.add(userRole);
+        createUser(testUser, password, new String[]{userRole}, ADMIN_LOGIN_USERNAME, ADMIN_PASSWORD);
+        userList.add(testUser);
+
+        apiPublisher.logout();
+        apiPublisher.login(testUser, password);
+        publishAPI(apiName, developer, userRole);
+
+        updateUser(developer, new String[]{"internal/subscriber"}, null, ADMIN_LOGIN_USERNAME, ADMIN_PASSWORD);
+        loginToStore(developer, password);
+        isAPIVisibleInStore(apiName, apiStoreClient);
+    }
+
+    @Test(description = "2.1.1.3", dataProvider = "ValidRoleDataProvider",
+            dataProviderClass = ScenarioDataProvider.class)
+    public void testPublishAlreadyPublishedAPIByValidRoleAssignedUser(String role) throws Exception {
+
+        apiName = "API" + count;
+        apiContext = "/verify" + count;
+        developer = "User_" + count;
+        testUser = "User" + count;
+        password = "password123$";
+        count++;
+
+        createUserWithPublisherAndCreatorRole(developer, password, ADMIN_LOGIN_USERNAME, ADMIN_PASSWORD);
+        apiPublisher.login(developer, password);
+        apiRequest = new APIRequest(apiName, apiContext, apiVisibility, apiVersion, apiResource, tierCollection,
+                new URL(backendEndPoint));
+        createAPI(apiRequest);
+        getAPI(apiName, developer, apiVersion);
+        apiNames.put(apiName, developer);
```

Comment:
Instead of repeating these pre-requisites in each test case, can't we add a @BeforeEach and do those at one place. If required we can clean those at @AfterEach

Reply:
The org.testng.annotations package which had been used for scenario testing is not included @beforeeach, @aftereach capability.